### PR TITLE
add cloud architecture and scaling docs

### PR DESCRIPTION
**Pull request by agola11**

### CODE REVIEW

File: versioned_docs/version-2.0/reference/cloud-arch.png
Comment (langchain-infra) [inline versioned_docs/version-2.0/reference/cloud-arch.png:1]: technically in gcp, we have several load balancers(separate api from frontend), not sure if we should mention?

File: versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- Google Cloud Storage (GCS) for runs inputs and outputs
+- Google Cloud SQL PostgreSQL for transactional workloads
+- Google Cloud Memorystore for Redis for queuing and caching
+- Clickhouse Cloud on GCP for trace ingestion and analytics. Our services connect to Clickhouse Cloud, which is hosted in the same GCP region, via a private endpoint.
+
+Some additional GCP services we use include:
+
+- Google Cloud Load Balancer for routing traffic to the LangSmith services
+- Google Cloud CDN for caching static assets
+- Google Cloud Armor for security
```
Comment (langchain-infra) [inline versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx:40]: do we have a rate limits section? maybe link to that?

**Review by langchain-infra — APPROVED** [review]
overall lgtm! just two comments/questions

File: versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+![](./cloud-arch.png)
+
+## Scalability
+
+LangSmith is designed to be scalable and performant.
+
+As of load testing done in February 2024, LangSmith can comfortably process 500K+ runs (spans) per minute.
+We anticipate that LangSmith can process 750K+ runs per minute with the optimizations we've made since then.
+
+![](./scalability.png)
```
Comment (akira) [inline versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx:51]: Maybe we can skip the diagram?  Would need to include more details if we wanted full diagrams / load tests

File: versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+## Architecture
+
+The US-based LangSmith service is deployed in the `us-central1` region of GCP.
+
+**NOTE:** We will be adding an EU instance deployed in `europe-west4` (Netherlands) soon. If you are interested in this, please contact us at sales@langchain.dev.
+
+LangSmith is composed of the following services, all deployed on Google Kubernetes Engine (GKE):
+
+- LangSmith Frontend
+- LangSmith Backend
```
Comment (akira) [inline versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx:24]: Any more detail on backend vs platform backend?

File: versioned_docs/version-2.0/reference/cloud-arch.png
Comment (agola11) [reply versioned_docs/version-2.0/reference/cloud-arch.png:1]: probs out of scope?

File: versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx (hunk 2 above)
Comment (agola11) [reply versioned_docs/version-2.0/reference/cloud_architecture_and_scalability.mdx:51]: yeah I can rm

**Review by akira — APPROVED** [review]

**Pull request merged**